apps/dashboard/app/apps/nav_bar.rb

```diff
@@ -0,0 +1,134 @@
+          matched_apps = nav_apps(nav_item, nil, nil)
+          extend_item(matched_apps.first.links.first, 'layouts/nav/link') if matched_apps.first && matched_apps.first.links.first
+        elsif nav_item.fetch(:template, nil)
+          nav_template(nav_item)
+        end
+      end
+    end.flatten.compact
+  end
+
+  private
+
+  def self.nav_menu(hash_item)
+    menu_title = hash_item.fetch(:title, '')
+    menu_items = hash_item.fetch(:links, [])
+
+    group_title = ''
+    apps = menu_items.map do |item|
+      group_title = item.fetch(:group, group_title)
+      if item.fetch(:url, nil)
+        nav_link(item, menu_title, group_title)
+      elsif item.fetch(:apps, nil)
+        nav_apps(item, menu_title, group_title)
+      elsif item.fetch(:profile, nil)
+        nav_profile(item, menu_title, group_title)
+      end
+    end.flatten.compact
+
+    OodAppGroup.new(apps: apps, title: menu_title, sort: false)
+  end
+
+  def self.nav_link(item, category, subcategory)
+    LinkDecorator.new(OodAppLink.new(item), category: category, subcategory: subcategory)
+  end
+
+  def self.nav_template(item)
+    template = File.join("layouts/nav", item.fetch(:template))
+    extend_item(item, template)
+  end
+
+  def self.nav_apps(item, category, subcategory)
+    app_configs = Array.wrap(item.fetch(:apps, []))
+    app_configs.map do |config_string|
+      matched_apps = Router.pinned_apps_from_token(config_string, SysRouter.apps)
+      matched_apps.map do |reg_app|
+        AppDecorator.new(reg_app, category: category, subcategory: subcategory)
+      end
+    end.flatten
+  end
+
+  def self.nav_profile(item, category, subcategory)
+    profile = item.fetch(:profile)
+    profile_data = item.clone
+    profile_data[:title] = profile unless item.fetch(:title, nil)
+    profile_data[:url] = Rails.application.routes.url_helpers.settings_path('settings[profile]' => profile)
+    profile_data[:data] = { method: 'post' }
+    profile_data[:new_tab] = false
+    LinkDecorator.new(OodAppLink.new(profile_data), category: category, subcategory: subcategory)
+  end
+
+  def self.item_from_token(token)
+    matched_apps = Router.pinned_apps_from_token(token, SysRouter.apps)
+    if matched_apps.size == 1
+      extend_item(matched_apps.first.links.first, 'layouts/nav/link')
+    elsif matched_apps.size > 1
+      extend_item(OodAppGroup.groups_for(apps: matched_apps), 'layouts/nav/group')
+    else
+      group = OodAppGroup.groups_for(apps: SysRouter.apps, sort: false).select { |g| g.title.downcase == token.downcase }.first
+      group.nil? ? nil : extend_item(group, 'layouts/nav/group')
+    end
+  end
+
+  def self.extend_item(item, partial_path)
+    extended = NavItemDecorator.new(item)
+    extended.partial_path = partial_path
+    extended
+  end
+
+  class NavItemDecorator < SimpleDelegator
+    attr_accessor :partial_path
+  end
+
+  # redefine an OodApp's category & subcategory
+  class AppDecorator < SimpleDelegator
```

Comment:
Can we go ahead and put this into it's own file? With it I think we can refactor FeaturedApps.

Maybe call it `app_recategorizer`?

Reply:
Are you saying to merge `AppDecorator` and `FeaturedApps` into a new class `AppRecategorizer`?
Or simply rename `FeaturedApp` to `AppRecategorizer` and creating a new file for `AppDecorator`?